torch/_guards.py

```diff
@@ -325,6 +326,18 @@ def get() -> Optional["TracingContext"]:
     def __init__(self, fake_mode):
         self.guards_context = GuardsContext()
         self.fake_mode = fake_mode
+        self.frame_summary_stack = []
+
+    @staticmethod
+    @contextlib.contextmanager
+    def current_frame(frame_summary):
+        tc = TracingContext.get()
+        assert (
+            tc is not None
+        ), "Frame context manager must be called within an ongoing trace."
+        tc.frame_summary_stack.append(frame_summary)
+        yield
+        TracingContext.get().frame_summary_stack.pop()
```

Comment:
This doesn't have to be here, it could be anywhere! But I kinda like having all our "global" state constrained to a single object with dynamic scoping. It beats having to deal with dynamo reset and thinking about cross frame logic.